src/standalone/Soil/rre.jl

```diff
@@ -504,9 +513,44 @@ end
 """
     is_saturated(Y, model::RichardsModel)
 
-A helper function which can be used to indicate whether a layer of soil is 
+A helper function which can be used to indicate whether a layer of soil is
 saturated.
 """
 function is_saturated(Y, model::RichardsModel)
     return @. ClimaLand.heaviside(Y.soil.ϑ_l - model.parameters.ν)
 end
+
+
+"""
+    add_dfluxBCdY_to_cache(p::NamedTuple,
+                            model::RichardsModel,
+                            bc::MoistureStateBC,
+                            ::ClimaLand.TopBoundary,
+                        )
+
+Allocate space in `p` for the derivative of the top boundary condition with
+respect to the state variables, and merges it into `p` under the key `dfluxBCdY`.
+Specifically, this stores the derivative of the top face flux with respect to
+the state variable at the top cell center.
+This is used in the implicit solver for
+`RichardsModel` when using a `MoistureStateBC` at the top boundary, in which
+case we need to handle the boundary specially in computing the Jacobian.
+
+"""
+# TODO should this go in p.dfluxBCdY or p.soil.dfluxBCdY?
+function ClimaLand.add_dfluxBCdY_to_cache(
```

Comment:
not sure if we need this method because we added the variable to the cache by adding it to `boundary_vars`. In this case, it is added to `p.soil`, not to `p` directly.